### PR TITLE
Novacustom nv4x/rel 1.2.0

### PR DESCRIPTION
**Pull request by mkopec**
First release based on the updated `dasharo-release` common branch.

- Rename `clevo_nv41mz` to `novacustom_nv4x` using the baseboard structure
- Fix nvidia driver compilation issue
- Fix s0ix suspend with certain SSDs
- Change BIOS vendor to `3mdeb` (this will change the fwupd GUID used for branch detection)

### CODE REVIEW

**Comment by miczyg1** [comment]
Why draft?

**Comment by mkopec** [comment]
@miczyg1 The YAMLs and build script aren't done yet and we're not releasing 1.2.0 yet, this is just preparation

**Comment by macpijan** [comment]
There is no reviewer assigned

File: configs/config.novacustom_nv4x [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+CONFIG_CONSOLE_CBMEM_BUFFER_SIZE=0x30000
+CONFIG_MAINBOARD_SMBIOS_PRODUCT_NAME="NV4XMB,ME,MZ"
+CONFIG_UART_PCI_ADDR=0x0
+CONFIG_PCIEXP_HOTPLUG_PREFETCH_MEM_BELOW_4G=y
+CONFIG_PCIEXP_HOTPLUG_IO=0x2000
+CONFIG_SUBSYSTEM_VENDOR_ID=0x0000
+CONFIG_SUBSYSTEM_DEVICE_ID=0x0000
+CONFIG_SMMSTORE_V2=y
+CONFIG_SMMSTORE_SIZE=0x40000
+# CONFIG_TPM_PPI is not set
```
Comment (miczyg1) [inline configs/config.novacustom_nv4x:20]: Why PPI is not set? On MSI I noticed it helped with get-tpm command on Windows somehow.
I have added PPI support to our payload not so long ago.

File: configs/config.novacustom_nv4x [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+CONFIG_CBFS_MCACHE_RW_PERCENTAGE=50
+CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0=y
+CONFIG_POST_IO_PORT=0x80
+CONFIG_PAYLOAD_TIANOCORE=y
+CONFIG_TIANOCORE_CUSTOM=y
+CONFIG_TIANOCORE_REPOSITORY="https://github.com/Dasharo/edk2.git"
+CONFIG_TIANOCORE_UEFIPAYLOAD=y
+CONFIG_TIANOCORE_TAG_OR_REV="4d2846baa98c253eadfb4c4e3085138e0d1396ab"
```
Comment (miczyg1) [inline configs/config.novacustom_nv4x:28]: This revision has TPM PPI support: cad237255be0eed3b184e927871791057d8da9db

File: dasharo/clevo_nv41mz_v1.0.0.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
+client:
```
Comment (miczyg1) [inline dasharo/clevo_nv41mz_v1.0.0.yaml:1]: Is this file actually needed? It may be always generated from v1.0.0 tag.

File: dasharo/project_config.yaml [hunk 4]
```diff
@@ -1,40 +1,44 @@
...
 client:
-  name: NovaCustom
+        name: NovaCustom
 
 framework:
-  name: coreboot
-  version: 4.16
-  url_template: https://github.com/Dasharo/coreboot/tree/COMMIT_HASH
+        name: coreboot
+        version: ae9a8447
```
Comment (miczyg1) [inline dasharo/project_config.yaml:14]: This is the latest upstream commit which the code is based on?

File: dasharo/releases.yaml [hunk 5]
```diff
@@ -1,24 +1,196 @@
 releases:
-        - version: v1.0.0
+        - version: v0.1.2
```
Comment (miczyg1) [inline dasharo/releases.yaml:2]: Newer versions should be always on top. Please reverse the order.

**Review by miczyg1 — CHANGES_REQUESTED** [review]

File: configs/config.novacustom_nv4x (hunk 1 above)
Comment (mkopec) [reply configs/config.novacustom_nv4x]: Done, although TPM seems to work fine without it on NV4x

File: configs/config.novacustom_nv4x (hunk 2 above)
Comment (mkopec) [reply configs/config.novacustom_nv4x]: Done

File: dasharo/clevo_nv41mz_v1.0.0.yaml (hunk 3 above)
Comment (mkopec) [reply dasharo/clevo_nv41mz_v1.0.0.yaml]: No, accidentaly commited. Removed

File: dasharo/project_config.yaml (hunk 4 above)
Comment (mkopec) [reply dasharo/project_config.yaml]: No, fixed.

File: dasharo/releases.yaml (hunk 5 above)
Comment (mkopec) [reply dasharo/releases.yaml]: Fixed

File: configs/config.novacustom_nv4x (hunk 2 above)
Comment (mkopec) [reply configs/config.novacustom_nv4x]: Seems to work fine with this change. There's a `ppi` entry in `/sys/class/tpm/tpm0` now

File: configs/config.novacustom_nv4x (hunk 1 above)
Comment (miczyg1) [reply configs/config.novacustom_nv4x]: Yes it should.

**Comment by miczyg1** [comment]
@mkopec `Change BIOS vendor to 3mdeb (this will change the fwupd GUID used for branch detection)` I don't see it in this PR  and it seemsto not be changed.

**Comment by mkopec** [comment]
@miczyg1 I accidentally the commit. Fixed

**Review by miczyg1 — APPROVED** [review]
